src/pathfinding_service/model/token_network.py

```diff
@@ -148,17 +154,18 @@ def _get_address_metadata(
         # Check node reachabilities
         metadata: Dict[str, Dict[str, Union[str, PeerCapabilities]]] = {}
         for node in self.nodes:
-            node_user_ids = self.reachability_state.get_userids_for_address(node)
+            node_users = self._user_manager.get_users_from_address(node)
             checksummed_address = to_checksum_address(node)
-            for user_id in node_user_ids:
-                if self.reachability_state.get_userid_presence(user_id) in [
+            for user in node_users:
+                if self.reachability_state.get_userid_presence(user.user_id) in [
                     UserPresence.ONLINE,
                     UserPresence.UNAVAILABLE,
                 ]:
                     capabilities = self.reachability_state.get_address_capabilities(node)
                     metadata[checksummed_address] = {
-                        "user_id": user_id,
+                        "user_id": user.user_id,
                         "capabilities": capabilities,
+                        "displayname": user.displayname or user.get_display_name(),
```

Comment:
`user.get_displayname()` implicitly returns `user.displayname` if it is set. So you calling the first is sufficient.